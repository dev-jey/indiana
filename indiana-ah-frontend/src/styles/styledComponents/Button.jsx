import styled from 'styled-components';

const Button = styled.button`
  background: ${(props) => {
    if (props.danger) {
      return '#CD160B';
    }
    if (props.bgColor) {
      return '#0B41CD';
    }
    return '#fff';
  }}
  font-size: 2rem;
  line-height: 2.7rem;
  align: Center;
  letterspacing: 5%;
  margin: ${props => (props.margin ? props.margin : '')};
  height: ${props => (props.height ? props.height : '')};
  border-radius: 0.4rem;
  text-align: center;
  color: ${(props) => {
    if (props.danger) {
      return '#fff';
    }
    if (props.bgColor) {
      return '#fff';
    }
    return '#0B41CD';
  }}
  cursor: pointer;
  border: ${(props) => {
    if (props.danger) {
      return '0.2rem solid #CD160B';
    }
    if (props.bgColor) {
      return 'none';
    }
    return '0.2rem solid #0B41CD';
  }}
  text-transform: capitalize;
  transition: all 0.3s;
  width: ${props => (props.width ? props.width : 'auto')};
  padding: ${props => (props.sm ? '.4rem 2.5rem' : '.8rem 1.3rem')};
  display: ${props => (props.inlineButton ? 'inline-block' : 'block')};
  :disabled {
    color: #dddddd;
    border: 0.2rem solid #dddddd;
  }
  @media (max-width: 830px) {
    width: auto;
    line-height: 2rem;
    font-size: 1.7rem;
  }
  @media (max-width: 355px) {
    width: auto;
    line-height: 1.5rem;
    padding: ${props => (props.sm ? '.4rem 1.5rem' : '.8rem 1.1rem')};
  }
`;

export default Button;
