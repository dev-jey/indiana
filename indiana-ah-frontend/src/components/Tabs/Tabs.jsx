import React, { Component } from 'react';
import { Tabs, Tab, Row } from 'react-bootstrap';
import Switch from 'react-switch';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Button from '../../styles/styledComponents/Button.jsx';
import Profileform from '../forms/profileForm.jsx';
import Passwordform from '../forms/passwordForm.jsx';
import { formatDate } from '../../utils/index';
import { updateInAppNotification } from '../../redux/actions/profileActions';

export class ProfileTab extends Component {
  state = {
    checked: this.props.userProfile.inAppNotification
  };

  handleChange = () => {
    this.setState({
      checked: !this.state.checked
    });
    this.props.updateInAppNotifications();
  };

  render() {
    const {
      userProfile: {
        bio, email, createdAt, name, username
      }
    } = this.props;
    return (
      <Tabs defaultActiveKey="Personal-Details" id="uncontrolled-tab-example">
        <Tab eventKey="Personal-Details" title="Personal Details">
          <div>
            <Row className="personal-details">
              <div className="col-sm-4 col-md-4">
                <h6 className="pb-15">Name</h6>
                <h5 className="black">{name || 'None'}</h5>
              </div>
              <div className="col-sm-4 col-md-4">
                <h6 className="pb-15">Email</h6>
                <h5 className="black">{email}</h5>
              </div>
              <div className="col-sm-4 col-md-4">
                <h6 className="pb-15">Date Registered</h6>
                <h5 className="black">{formatDate(createdAt)}</h5>
              </div>
            </Row>
            <Row className="bio">
              <div className="col-md-11">
                <h6 className="pb-15">Bio</h6>
                <h5 className="black bio-text">{bio || 'None'}</h5>
              </div>
            </Row>
          </div>
        </Tab>
        <Tab eventKey="Update-Details" title="Update Details">
          <Row className="w-100">
            <div className="update-details">
              <Profileform name={name} bio={bio} username={username} />
              <Passwordform />
            </div>
          </Row>
        </Tab>
        <Tab eventKey="Settings" title="Settings">
          <Row className="settings d-flex">
            <div className="col-md-6 notification-box text-center">
              <h3>Turn on Notifications</h3>
              <p>
                Subscribing to this service will alert you to events such as someone
                liking your article
              </p>
              <Switch
                onChange={this.handleChange}
                checked={this.state.checked || false}
              />
              {this.state.checked ? (
                <h3 className="mt-2 mb-0">Turn Off</h3>
              ) : (
                <h3 className="mt-2 mb-0">Turn On</h3>
              )}
            </div>
            <div className="col-md-6 deactivate text-center">
              <h3>Deactivate Your Account</h3>
              <p>
                Once Deactivated, you will not be able to access your account Ever Again
              </p>
              <Button danger sm>
                Deactivate
              </Button>
            </div>
          </Row>
        </Tab>
      </Tabs>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  updateInAppNotifications: () => dispatch(updateInAppNotification())
});

ProfileTab.propTypes = {
  updateInAppNotifications: PropTypes.func,
  userProfile: PropTypes.object,
  user: PropTypes.object
};

export default connect(null, mapDispatchToProps)(ProfileTab);
