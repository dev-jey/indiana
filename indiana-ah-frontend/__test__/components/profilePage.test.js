import React from 'react';
import { mount, shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Button from '../../src/styles/styledComponents/Button.jsx';
import ConnectedProfilePage, { ProfilePage } from '../../src/components/ProfilePage.jsx';
import { ProfileTab } from '../../src/components/Tabs/Tabs.jsx';
import Profileform, { ProfileForm } from '../../src/components/forms/profileForm.jsx';
import Passwordform, { PasswordForm } from '../../src/components/forms/passwordForm.jsx';

const onChange = jest.fn();
const initialState = {
  user: {
    userData: {
      name: 'Ezenwa Ogbonna',
      id: 'dac4d2c6-2d27-49da-a3c8-6e1c86518099',
      username: 'drizzy',
      email: 'ezenwaogbonna1@gmail.com',
      bio: 'jwkwewkwkkwkwwww',
      imageUrl: 'http://s3.amazonaws.com/37assets/svn/765-default-avatar.png',
      isVerified: true,
      role: 'user',
      subscribed: true,
      inAppNotification: false,
      createdAt: '2019-03-25T12:44:11.396Z'
    },
    isLoading: false,
    error: 'username error'
  }
};

const mockStore = configureStore([thunk]);
const store = mockStore(initialState);

describe('Profile page component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <ConnectedProfilePage />
          </Provider>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Profile page', () => {
  let profilePage;
  beforeEach(() => {
    profilePage = mount(
      <MemoryRouter>
        <Provider store={store}>
          <ConnectedProfilePage />
        </Provider>
      </MemoryRouter>
    );
  });
  afterEach(() => {
    profilePage.unmount();
  });
  it('renders the Profile Page', () => {
    expect(profilePage.find('main')).toBeDefined();
    expect(profilePage.find(<ProfileTab />)).toBeDefined();
  });
  it('toggles the checkbox', () => {
    profilePage.instance();
    const checkbox = profilePage.find('input').at(1);
    checkbox.simulate('change', { target: { checked: true } });
  });
  it('calls the handleProfileImage functions', () => {
    profilePage
      .find('input')
      .at(0)
      .simulate('change');
  });
  it('should display the apporpriate card when the user has not completed his/her profile', () => {
    const props = {
      userData: {
        bio: '',
        name: 'dodo',
        username: 'dododo',
        subscribed: true
      },
      isLoading: true
    };
    const wrapper = shallow(<ProfilePage user={props} />);
    expect(
      wrapper
        .find('h5')
        .at(2)
        .text()
    ).toEqual('Your account has been verified');
  });

  describe('Update Password form', () => {
    const handleSubmit = jest.fn();
    it('should change state', () => {
      const passwordForm = mount(
        <Provider store={store}>
          <Passwordform handleSubmit={handleSubmit} handleChange={onChange} />
        </Provider>
      );
      const input = passwordForm.find('input').at(0);
      input.instance().value = 'cim23000';
      input.simulate('blur');
      input.simulate('change');
      passwordForm.setState({ currentPassword: 'cim23000' });
      expect(passwordForm.state().currentPassword).toEqual('cim23000');
      const button = passwordForm.find('button');
      button.simulate('click');
    });
  });

  describe('Update Profile form', () => {
    const handleSubmit = jest.fn();
    it('should change state', () => {
      const profileForm = mount(
        <Provider store={store}>
          <Profileform handleSubmit={handleSubmit} handleChange={onChange} />
        </Provider>
      );
      const input = profileForm.find('input').at(0);
      input.instance().value = 'Yinka Alabi';
      input.simulate('change');
      profileForm.setState({ name: 'Yinka Alabi' });
      expect(profileForm.state().name).toEqual('Yinka Alabi');
      const button = profileForm.find('button');
      button.simulate('click');
    });
    it('should handle errors from profile update', () => {
      const profileData = {
        error: 'error in username oo'
      };
      const userProfile = {
        bio: 'yeye yeye',
        name: 'dodo',
        username: 'dododo',
        isLoading: true
      };
      const wrapper = shallow(
        <ProfileForm
          profileData={profileData}
          handleSubmit={handleSubmit}
          handleChange={onChange}
          userProfile={userProfile}
        />
      );
      wrapper.setProps({ profileData: { error: 'error in username sha' } });
      expect(wrapper.state().errors.username).toEqual('error in username sha');
    });
    it('should handle password validation errors', () => {
      const profileData = {
        error: 'error in username oo'
      };
      const updateUserPassword = jest.fn();
      const userProfile = {
        bio: 'yeye yeye',
        name: 'dodo',
        username: 'dododo',
        isLoading: true
      };
      const wrapper = shallow(
        <PasswordForm
          profileData={profileData}
          updateUserPassword={updateUserPassword}
          handleSubmit={handleSubmit}
          handleChange={onChange}
          userProfile={userProfile}
        />
      );
      wrapper.setProps({ profileData: { error: 'Passwords don\'t match' } });
      wrapper.setState({
        data: {
          currentPassword: 'dodosecret',
          newPassword: 'dadasecret',
          confirmPassword: 'dadasecreta'
        }
      });
      const wrongPasswordEvent = {
        preventDefault() { },
        target: {
          currentPassword: 'dodosecret',
          newPassword: 'dadasecret',
          confirmPassword: 'dadasecreta'
        }
      };
      wrapper.instance().handleBlur(wrongPasswordEvent);
      expect(wrapper.state().errors.confirmPassword).toEqual('Passwords don\'t match');
      wrapper.setState({
        data: {
          currentPassword: 'dodosecret',
          newPassword: 'dadasecret',
          confirmPassword: 'dadasecret'
        }
      });
      const correctPasswordEvent = {
        preventDefault() { },
        target: {
          currentPassword: 'dodosecret',
          newPassword: 'dadasecret',
          confirmPassword: 'dadasecret'
        }
      };
      wrapper.instance().handleBlur(correctPasswordEvent);
      expect(wrapper.state().errors.confirmPassword).toEqual('');
      wrapper.setState({
        data: {
          currentPassword: 'dodosecret',
          newPassword: 'dadasecret',
          confirmPassword: 'dadasecret'
        }
      });
      wrapper
        .find(Button)
        .dive()
        .simulate('click');
      expect(wrapper.state().errors.confirmPassword).toEqual('');
    });
  });
});
