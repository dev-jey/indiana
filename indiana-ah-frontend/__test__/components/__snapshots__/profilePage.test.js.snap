// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile page component renders correctly 1`] = `
<div
  className="main"
>
  <div
    className="sidenav"
  >
    <a
      aria-current={null}
      className="sidenav__logo-box"
      href="/profile"
      onClick={[Function]}
    >
      <img
        className="sideNavLogos"
        src="test-file-stub"
      />
      <span
        className="sidenav__logo-label"
      >
        Profile
      </span>
    </a>
    <a
      aria-current={null}
      className="sidenav__logo-box"
      href="/dashboard/posts"
      onClick={[Function]}
    >
      <img
        className="sideNavLogos"
        src="test-file-stub"
      />
      <span
        className="sidenav__logo-label"
      >
        Articles
      </span>
    </a>
    <a
      aria-current={null}
      className="sidenav__logo-box"
      href="/statistics"
      onClick={[Function]}
    >
      <img
        className="sideNavLogos"
        src="test-file-stub"
      />
      <span
        className="sidenav__logo-label"
      >
        Statistics
      </span>
    </a>
  </div>
  <section
    className="container-fluid profile"
  >
    <div
      className="mb-5 row"
    >
      <div
        className="col-md-4 col-sm-5 profile-card text-center profile-info"
      >
        <h5>
          Profile
        </h5>
        <div>
          <img
            alt="Profile Picture"
            className="profile-pic"
            src="http://s3.amazonaws.com/37assets/svn/765-default-avatar.png"
          />
          <span>
            <label
              htmlFor="file"
            >
              <img
                alt="camera"
                className="camera"
                src="test-file-stub"
              />
            </label>
            <input
              id="file"
              onChange={[Function]}
              type="file"
            />
          </span>
        </div>
        <h5
          className="black"
        >
          drizzy
        </h5>
      </div>
      <div
        className="space-sm col-md-1 col-sm-1"
      />
      <div
        className="col-md-7 col-sm-6 profile-card finished"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-4 profile-card text-center mail order-3 order-md-4"
      >
        <div>
          <img
            alt="mail icon"
            className="d-block m-auto mail-ico"
            src="test-file-stub"
          />
        </div>
        <h3
          className="mail-text"
        >
          Subscribe To Email Notification
        </h3>
        <h6>
          To recieve app notifications in your email
        </h6>
        <div
          className="subscribe"
        >
          <form>
            <input
              checked={true}
              className="mr-3"
              id="subscribe"
              name="subscribe"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="black"
              htmlFor="subscribe"
            >
              Subscribe
            </label>
          </form>
        </div>
      </div>
      <div
        className="space col-md-1 order-2 order-md-4"
      />
      <div
        className="col-md-7 tabs profile-card order-1 order-md-4"
      >
        <nav
          className="nav nav-tabs"
          onKeyDown={[Function]}
          role="tablist"
        >
          <a
            aria-controls="uncontrolled-tab-example-tabpane-Personal-Details"
            aria-selected={true}
            className="nav-item nav-link active"
            data-rb-event-key="Personal-Details"
            href="#"
            id="uncontrolled-tab-example-tab-Personal-Details"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="tab"
          >
            Personal Details
          </a>
          <a
            aria-controls="uncontrolled-tab-example-tabpane-Update-Details"
            aria-selected={false}
            className="nav-item nav-link"
            data-rb-event-key="Update-Details"
            href="#"
            id="uncontrolled-tab-example-tab-Update-Details"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="tab"
            tabIndex={-1}
          >
            Update Details
          </a>
          <a
            aria-controls="uncontrolled-tab-example-tabpane-Settings"
            aria-selected={false}
            className="nav-item nav-link"
            data-rb-event-key="Settings"
            href="#"
            id="uncontrolled-tab-example-tab-Settings"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="tab"
            tabIndex={-1}
          >
            Settings
          </a>
        </nav>
        <div
          className="tab-content"
        >
          <div
            aria-hidden={false}
            aria-labelledby="uncontrolled-tab-example-tab-Personal-Details"
            className="fade tab-pane active show"
            id="uncontrolled-tab-example-tabpane-Personal-Details"
            role="tabpanel"
          >
            <div>
              <div
                className="personal-details row"
              >
                <div
                  className="col-sm-4 col-md-4"
                >
                  <h6
                    className="pb-15"
                  >
                    Name
                  </h6>
                  <h5
                    className="black"
                  >
                    Ezenwa Ogbonna
                  </h5>
                </div>
                <div
                  className="col-sm-4 col-md-4"
                >
                  <h6
                    className="pb-15"
                  >
                    Email
                  </h6>
                  <h5
                    className="black"
                  >
                    ezenwaogbonna1@gmail.com
                  </h5>
                </div>
                <div
                  className="col-sm-4 col-md-4"
                >
                  <h6
                    className="pb-15"
                  >
                    Date Registered
                  </h6>
                  <h5
                    className="black"
                  >
                    25/3/2019
                  </h5>
                </div>
              </div>
              <div
                className="bio row"
              >
                <div
                  className="col-md-11"
                >
                  <h6
                    className="pb-15"
                  >
                    Bio
                  </h6>
                  <h5
                    className="black bio-text"
                  >
                    jwkwewkwkkwkwwww
                  </h5>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden={true}
            aria-labelledby="uncontrolled-tab-example-tab-Update-Details"
            className="fade tab-pane"
            id="uncontrolled-tab-example-tabpane-Update-Details"
            role="tabpanel"
          >
            <div
              className="w-100 row"
            >
              <div
                className="update-details"
              >
                <div
                  className="col-md-12 profile-form"
                >
                  <div
                    className="details-form"
                  >
                    <div
                      className="input-div"
                    >
                      <label
                        htmlFor="name"
                      >
                        Name
                      </label>
                      <input
                        className="sc-bwzfXH bdRrXL"
                        id="name"
                        name="name"
                        onChange={[Function]}
                        placeholder="Ezenwa Ogbonna"
                        required={true}
                        type="text"
                        value="Ezenwa Ogbonna"
                      />
                      <small
                        className="text-danger ml-3 form-text"
                      />
                    </div>
                    <div
                      className="input-div"
                    >
                      <label
                        htmlFor="username"
                      >
                        Username
                      </label>
                      <input
                        className="sc-bwzfXH lkMXIB"
                        id="username"
                        name="username"
                        onChange={[Function]}
                        placeholder="drizzy"
                        required={true}
                        type="text"
                        value="drizzy"
                      />
                      <small
                        className="text-danger ml-3 form-text"
                      />
                    </div>
                  </div>
                  <div
                    className="bio-form"
                  >
                    <div
                      className="form-group"
                    >
                      <label
                        className="form-label"
                        htmlFor="exampleForm.ControlTextarea1"
                      >
                        Bio
                      </label>
                      <textarea
                        className="form-control"
                        id="exampleForm.ControlTextarea1"
                        name="bio"
                        onChange={[Function]}
                        placeholder="jwkwewkwkkwkwwww"
                        rows="3"
                        value="jwkwewkwkkwkwwww"
                      />
                    </div>
                  </div>
                  <button
                    className="sc-bdVaJa grSjkW"
                    disabled={false}
                    onClick={[Function]}
                  >
                    Update
                  </button>
                </div>
                <div
                  className="col-md-6 password-form"
                >
                  <h3
                    className="text-center"
                  >
                    Change Password
                  </h3>
                  <div>
                    <div
                      className="input-div"
                    >
                      <label
                        htmlFor="currentPassword"
                      />
                      <input
                        className="sc-bwzfXH dwtOFN"
                        id="currentPassword"
                        name="currentPassword"
                        onBlur={[Function]}
                        onChange={[Function]}
                        placeholder="Current Password"
                        required={true}
                        type="password"
                        value=""
                      />
                      <small
                        className="text-danger ml-3 form-text"
                      >
                        
                      </small>
                    </div>
                  </div>
                  <div>
                    <div
                      className="input-div"
                    >
                      <label
                        htmlFor="newPassword"
                      />
                      <input
                        className="sc-bwzfXH dwtOFN"
                        id="newPassword"
                        name="newPassword"
                        onBlur={[Function]}
                        onChange={[Function]}
                        placeholder="New Password"
                        required={true}
                        type="password"
                        value=""
                      />
                      <small
                        className="text-danger ml-3 form-text"
                      >
                        
                      </small>
                    </div>
                  </div>
                  <div
                    className="mb-30"
                  >
                    <div
                      className="input-div mb-30"
                    >
                      <label
                        htmlFor="confirmPassword"
                      />
                      <input
                        className="sc-bwzfXH dwtOFN"
                        id="confirmPassword"
                        name="confirmPassword"
                        onBlur={[Function]}
                        onChange={[Function]}
                        placeholder="Confirm Password"
                        required={true}
                        type="password"
                        value=""
                      />
                      <small
                        className="text-danger ml-3 form-text"
                      />
                    </div>
                  </div>
                  <button
                    className="sc-bdVaJa grSjkW"
                    disabled={true}
                    onClick={[Function]}
                  >
                    Save
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden={true}
            aria-labelledby="uncontrolled-tab-example-tab-Settings"
            className="fade tab-pane"
            id="uncontrolled-tab-example-tabpane-Settings"
            role="tabpanel"
          >
            <div
              className="settings d-flex row"
            >
              <div
                className="col-md-6 notification-box text-center"
              >
                <h3>
                  Turn on Notifications
                </h3>
                <p>
                  Subscribing to this service will alert you to events such as someone liking your article
                </p>
                <div
                  style={
                    Object {
                      "MozTransition": "opacity 0.25s",
                      "MozUserSelect": "none",
                      "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                      "WebkitTransition": "opacity 0.25s",
                      "WebkitUserSelect": "none",
                      "borderRadius": 14,
                      "display": "inline-block",
                      "msUserSelect": "none",
                      "opacity": 1,
                      "position": "relative",
                      "textAlign": "left",
                      "touchAction": "none",
                      "transition": "opacity 0.25s",
                      "userSelect": "none",
                    }
                  }
                >
                  <div
                    className="react-switch-bg"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    style={
                      Object {
                        "MozTransition": "background 0.25s",
                        "WebkitTransition": "background 0.25s",
                        "background": "#888888",
                        "borderRadius": 14,
                        "cursor": "pointer",
                        "height": 28,
                        "margin": 0,
                        "position": "relative",
                        "transition": "background 0.25s",
                        "width": 56,
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "MozTransition": "opacity 0.25s",
                          "WebkitTransition": "opacity 0.25s",
                          "height": 28,
                          "opacity": 0,
                          "pointerEvents": "none",
                          "position": "relative",
                          "transition": "opacity 0.25s",
                          "width": 30,
                        }
                      }
                    >
                      <svg
                        height="100%"
                        style={
                          Object {
                            "position": "absolute",
                            "top": 0,
                          }
                        }
                        viewBox="-2 -5 17 21"
                        width="100%"
                      >
                        <path
                          d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
                          fill="#fff"
                          fillRule="evenodd"
                        />
                      </svg>
                    </div>
                    <div
                      style={
                        Object {
                          "MozTransition": "opacity 0.25s",
                          "WebkitTransition": "opacity 0.25s",
                          "height": 28,
                          "opacity": 1,
                          "pointerEvents": "none",
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                          "transition": "opacity 0.25s",
                          "width": 30,
                        }
                      }
                    >
                      <svg
                        height="100%"
                        style={
                          Object {
                            "position": "absolute",
                            "top": 0,
                          }
                        }
                        viewBox="-2 -5 14 20"
                        width="100%"
                      >
                        <path
                          d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
                          fill="#fff"
                          fillRule="evenodd"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    className="react-switch-handle"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onTouchCancel={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "MozTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                        "WebkitTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                        "background": "#ffffff",
                        "border": 0,
                        "borderRadius": "50%",
                        "boxShadow": null,
                        "cursor": "pointer",
                        "display": "inline-block",
                        "height": 26,
                        "outline": 0,
                        "position": "absolute",
                        "top": 1,
                        "transform": "translateX(1px)",
                        "transition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                        "width": 26,
                      }
                    }
                  />
                  <input
                    checked={false}
                    disabled={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyUp={[Function]}
                    role="switch"
                    style={
                      Object {
                        "border": 0,
                        "clip": "rect(0 0 0 0)",
                        "height": 1,
                        "margin": -1,
                        "overflow": "hidden",
                        "padding": 0,
                        "position": "absolute",
                        "width": 1,
                      }
                    }
                    type="checkbox"
                  />
                </div>
                <h3
                  className="mt-2 mb-0"
                >
                  Turn On
                </h3>
              </div>
              <div
                className="col-md-6 deactivate text-center"
              >
                <h3>
                  Deactivate Your Account
                </h3>
                <p>
                  Once Deactivated, you will not be able to access your account Ever Again
                </p>
                <button
                  className="sc-bdVaJa jZpGVp"
                >
                  Deactivate
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
